### PR TITLE
chore(get_current_session): Return default session if current is explicitly None

### PR DESCRIPTION
**Pull request by gadenbuie**
From https://github.com/posit-dev/py-shiny/pull/1084#issuecomment-1915683032

### CODE REVIEW

**Review by wch — APPROVED** [review]
Looks good! But let's hold off merging for a bit, until after we merge in various post-0.7.0 fixes and do a small release.

**Comment by gadenbuie** [comment]
@wch should we merge this?

**Comment by wch** [comment]
Yes, good idea, thanks!

**Pull request merged**
